src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/PosixSignalRegistration.Unix.cs

```diff
@@ -9,9 +9,9 @@ namespace System.Runtime.InteropServices
 {
     public sealed partial class PosixSignalRegistration
     {
-        private static readonly Dictionary<int, HashSet<Token>> s_registrations = Initialize();
+        private static readonly Dictionary<int, List<Token>> s_registrations = Initialize();
```

Comment:
`List<Token>` will require O(N) linear search for unregistration. I do not expect it to be a problem since the number of these registrations is expected to be fairly small.